ideascube/library/migrations/0008_library_stock.py

```diff
@@ -23,16 +27,45 @@ def migrate_data(apps, schema_editor):
             subtitle=old_book.subtitle, publisher=old_book.publisher,
             section=old_book.section, lang=old_book.lang, cover=old_book.cover,
             tags=old_book.tags)
-        new_book.save()
+        old_specimen = None
+        new_bookspecimens = []
 
         for old_bookspecimen in old_book.specimens.all():
-            new_bookspecimen = BookSpecimen(
-                barcode=old_bookspecimen.serial, item=new_book, count=1,
-                comments=old_bookspecimen.remarks,
-                # And now the stuff which hasn't changed
-                location=old_bookspecimen.location, file=old_bookspecimen.file)
-            new_bookspecimen.save()
+            if old_bookspecimen.serial in specimens:
+                # If "old serial" equals "new barcode" then we are dealing with
+                # the same specimen
+                old_specimen = specimens.pop(old_bookspecimen.serial)
+
+            if old_specimen is not None:
+                # Either we just got old_specimen in the previous if condition,
+                # or we got it on a previous iteration
+
+                if new_book.id is None:
+                    new_book.stockitem_ptr = old_specimen.item
+                    new_book.id = old_specimen.item.id
+
+                new_bookspecimen = BookSpecimen(
+                    specimen_ptr=old_specimen,
+                    # And now the stuff which hasn't changed
+                    location=old_bookspecimen.location,
+                    file=old_bookspecimen.file)
```

Comment:
Do we've lost the (BookSpeciment) comments / (oldBookSpeciment) remarks ?
The remarks become the comments in the migration. You take this into account when you create the BookSpecimen line 55, but not here.